### ISSUE TITLE
Broken HighDPI scaling on Plasma

### ISSUE THREAD

**Issue by nicolasfella** [ksnip/kImageAnnotator#326]
UI elements in Gwenview's annotation tool (which uses kImageAnnotator) are too large when using HighDPI scaling. See https://bugs.kde.org/show_bug.cgi?id=479215

It worked fine until https://github.com/ksnip/kImageAnnotator/pull/302, which is conceptually wrong. Qt already does proper scaling out of the box. Manually multiplying a scale factor onto things is wrong in most cases and causes things to be scaled twice.

**Comment by mkurz** [comment]
Anyone?

**Comment by mkurz** [comment]
I can also confirm applying the revert (#339) fixes the problem for me.

**Comment by DamirPorobic** [comment]
> > ... which is conceptually wrong. Qt already does proper scaling out of the box. Manually multiplying a scale factor onto things is wrong in most cases and causes things to be scaled twice.
> 
> [@nicolasfella](https://github.com/nicolasfella) it seems you know Qt - can you point to some docs to what you say here? Thanks!

Yes, conceptually one would expect Qt to do the work but the truth is they have bugs, not everything works as expected. I have reported some of the bugs myself to Qt and I haven't seen any one of them fixed so I had to implement workarounds. Might be that those are fixed now (in later versions) and you see that we are now overscaling.

**Comment by mkurz** [comment]
@DamirPorobic This can be closed I guess.
